### PR TITLE
Add a new command to get the issues filtered by their state

### PR DESCRIPTION
**Pull request by NotBad4U**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -32,11 +32,28 @@ use std::error::Error;
...
+struct MyIssueState(IssueState);
+
+impl FromStr for MyIssueState {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_ref() {
+            "open" => Ok(MyIssueState(IssueState::Opened)),
+            "close" => Ok(MyIssueState(IssueState::Closed)),
+            "reopen" => Ok(MyIssueState(IssueState::Reopened)),
```
Comment (divarvel) [inline src/main.rs]: can you rename it to `reopened` to be consistent with the enum?

File: src/main.rs [hunk 2]
```diff
@@ -32,11 +32,28 @@ use std::error::Error;
...
+            _ => Err(format!("Unknow state: {}", s)),
```
Comment (divarvel) [inline src/main.rs]: s/Unknow/Unknown/

File: src/main.rs [hunk 3]
```diff
@@ -181,6 +198,24 @@ fn get_user_id_by_name(name: &str) -> Result<UserId, Box<Error>> {
...
+    filter_state: &MyIssueState,)
+    -> Result<String, Box<Error>> {
+    let gitlab_client = Gitlab::new(config.gitlab_domain, config.gitlab_token)?;
+    let project = gitlab_client.project_by_name(project)?;
+
+    gitlab_client.issues(project.id)
+        .and_then(|issues| {
+            issues.into_iter()
+            .filter(|i| i.state == filter_state.0)
+            .for_each(|i| println!("#{} {:?} {} {}", i.id, i.state, i.title, i.created_at.format("%F %H:%M")));
```
Comment (divarvel) [inline src/main.rs]: Not sure about how it's usually done in Rust, but relying on `Debug` for regular display seems a bit odd.
Poke @Geal

File: src/main.rs [hunk 4]
```diff
@@ -287,6 +329,12 @@ enum Cmd {
         text: Option<String>,
     },
     #[structopt(name = "init", about = "Generate configuration")] Init {},
+    #[structopt(name = "l", about = "List all gitlab issues")]
+    ListIssues {
+        #[structopt(name = "filter", short = "f", long = "filter",
+                    help = "Filter the issues by state. Possible values are: open, close, reopen")]
```
Comment (divarvel) [inline src/main.rs]: see first comment about "reopened"

**Review by divarvel — CHANGES_REQUESTED** [review]
Hi, apart from a few typos here and there and the use of Debug for regular display, LGTM

File: src/main.rs (hunk 3 above)
Comment (NotBad4U) [reply src/main.rs]: I can implement the `Trait Display` for `MyIssueState` to correct this if you want ?

Comment (Geal) [reply src/main.rs]: yes, `Debug` is not meant for the user side

File: src/main.rs [hunk 5]
```diff
@@ -32,11 +32,49 @@ use std::error::Error;
...
+#[derive(Debug)]
+struct MyIssueState(IssueState);
+
+impl FromStr for MyIssueState {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_ref() {
+            "open" => Ok(MyIssueState(IssueState::Opened)),
+            "close" => Ok(MyIssueState(IssueState::Closed)),
```
Comment (divarvel) [inline src/main.rs]: s/close/closed/

File: src/main.rs [hunk 6]
```diff
@@ -32,11 +32,49 @@ use std::error::Error;
...
+            IssueState::Reopened => MyIssueState(IssueState::Reopened),
+        }
+    }
+}
+
+impl fmt::Display for MyIssueState {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self.0 {
+            IssueState::Opened => write!(f, "open"),
+            IssueState::Closed => write!(f, "close"),
```
Comment (divarvel) [inline src/main.rs]: s/close/closed/

File: src/main.rs [hunk 7]
```diff
@@ -287,6 +350,12 @@ enum Cmd {
         text: Option<String>,
     },
     #[structopt(name = "init", about = "Generate configuration")] Init {},
+    #[structopt(name = "l", about = "List all gitlab issues")]
+    ListIssues {
+        #[structopt(name = "filter", short = "f", long = "filter",
+                    help = "Filter the issues by state. Possible values are: open, close, reopened")]
```
Comment (divarvel) [inline src/main.rs]: s/close/closed/

**Review by divarvel — CHANGES_REQUESTED** [review]

**Comment by divarvel** [comment]
I'm fixing a few issues and adding some bells and whistles.

**Review by divarvel — APPROVED** [review]

**Comment by divarvel** [comment]
@NotBad4U I've fixed the typos, added a default value for `--filter`, displayed the URLs in the list output. Looks good to you?

**Comment by NotBad4U** [comment]
Look good to me too 👍

**Pull request merged**